File: 프로그래머스/Lv.0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
         
        int denom=denom1*denom2;
        int numer= denom1*numer2+denom2*numer1 ;
        int max=1;
        //분자, 분모의 최대공약수를 구해봅시다
        for(int i=1; i<=denom&&i<=numer; i++ ){
            if(numer%i==0 && denom%i==0){
                max=i;
            }
        }
        numer=numer/max;
        denom=denom/max;
        int[] answer={numer,denom};
        
        return answer;
    }
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.kt
class Solution {
    fun solution(numer1: Int, denom1: Int, numer2: Int, denom2: Int): IntArray {
        var answer: IntArray = intArrayOf(0,0)
        val denom: Int = denom1 *  denom2
        val numer: Int = denom1 * numer2 + denom2 * numer1
        
        for(i in denom downTo 1){
            if(denom % i == 0 && numer % i == 0 ){
                answer[0] = numer / i
                answer[1] = denom / i
                break;
            }
        }
        
        return answer
    }
}